### PR TITLE
Add thumb color customization to CupertinoSwitch

### PR DESCRIPTION
**Pull request by sunyujia-rakuten**
## Description
When using the CupertinoSwitch allow to set the thumb color. 

This fixed the following issue: #86774

## Tests
I added the following tests:

* A test which confirms that the CupertinoSwitch.thumbColor is set
* A test which confirms that the CupertinoSwitch.thumbColor is using default

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

 ## Breaking Change
Does your PR require Flutter developers to manually update their apps to accommodate your change?
- [ ] Yes, this is a breaking change (Please read Handling breaking changes). Replace this with a link to the e-mail where you asked for input on this proposed change.
- [x] No, this is not a breaking change.

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 2 new digest(s) on patchset 1.
View them at https://flutter-gold.skia.org/cl/github/86775

File: packages/flutter/lib/src/cupertino/switch.dart [hunk 1]
```diff
@@ -428,6 +441,16 @@ class _RenderCupertinoSwitch extends RenderConstrainedBox {
     markNeedsPaint();
   }
 
+  Color get thumbColor => _thumbColor;
+  Color _thumbColor;
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/cupertino/switch.dart:445]: nit: alternatively you can hold the `CupertinoThumbPainter` in an instance variable and `thumbColor` is just `thumbPainter.color`.

File: packages/flutter/test/cupertino/switch_test.dart [hunk 2]
```diff
@@ -526,6 +526,43 @@ void main() {
     expect(find.byType(CupertinoSwitch), paints..rrect(color: trackColor));
   });
 
+  testWidgets('Switch is using default thumb color', (WidgetTester
+  tester) async {
```
Comment (LongCatIsLooong) [inline packages/flutter/test/cupertino/switch_test.dart:530]: This probably should be on the same line as `WidgetTester`.

File: packages/flutter/test/cupertino/switch_test.dart [hunk 3]
```diff
@@ -526,6 +526,43 @@ void main() {
...
+          child: CupertinoSwitch(
+            value: false,
+            onChanged: null,
+          ),
+        ),
+      ),
+    );
+
+    expect(find.byType(CupertinoSwitch), findsOneWidget);
+    expect(tester.widget<CupertinoSwitch>(find.byType(CupertinoSwitch)).thumbColor, CupertinoColors.white);
```
Comment (LongCatIsLooong) [inline packages/flutter/test/cupertino/switch_test.dart:544]: Could you also check the thumb is painted in the right color, similar to line 526?

Comment (itssunyujia) [reply packages/flutter/test/cupertino/switch_test.dart:544]: I planned to do this, but I don’t know how to test the thumb painter. Is there an example that I can refer to? thank you!

File: packages/flutter/test/cupertino/switch_test.dart (hunk 2 above)
Comment (itssunyujia) [reply packages/flutter/test/cupertino/switch_test.dart:530]: I will fix this

File: packages/flutter/test/cupertino/switch_test.dart (hunk 3 above)
Comment (itssunyujia) [reply packages/flutter/test/cupertino/switch_test.dart:544]: Thanks for reminding, I will add thumb paint test.

File: packages/flutter/test/cupertino/switch_test.dart (hunk 2 above)
Comment (itssunyujia) [reply packages/flutter/test/cupertino/switch_test.dart]: done!

File: packages/flutter/test/cupertino/switch_test.dart (hunk 3 above)
Comment (itssunyujia) [reply packages/flutter/test/cupertino/switch_test.dart]: done!

File: packages/flutter/lib/src/cupertino/switch.dart (hunk 1 above)
Comment (itssunyujia) [reply packages/flutter/lib/src/cupertino/switch.dart:445]: I'm not sure that I comprehend your meaning ,so i add some codes below.

Comment (LongCatIsLooong) [reply packages/flutter/lib/src/cupertino/switch.dart:445]: Yeah that's what I meant.

**Comment by LongCatIsLooong** [comment]
6a6c34d LGTM but you might need to remove the irrelevant commits.

**Comment by itssunyujia** [comment]
> [6a6c34d](https://github.com/flutter/flutter/commit/6a6c34d0ee16641b5fe87c33b12b9e7754a470b4) LGTM but you might need to remove the irrelevant commits.

yeah my bad

**Review by LongCatIsLooong — APPROVED** [review]
LGTM. Thanks for contributing!

**Comment by itssunyujia** [comment]
> LGTM. Thanks for contributing!

Hi , @LongCatIsLooong . I don't  remember how I trigger this review request to @jmagman  for this PR .  Is there anything else I need to do?

**Comment by jmagman** [comment]
> Is there anything else I need to do?

This needed the `waiting for tree to go green` label added to merge.  Added the label, should go once the tree reopens.

**Comment by itssunyujia** [comment]
> This needed the waiting for tree to go green label added to merge. Added the label, should go once the tree reopens.

OKey，Thanks！

**Pull request merged**